packages/react-components/react-drawer/src/components/Drawer/renderDrawer.tsx

```diff
@@ -8,6 +8,17 @@ import type { DrawerState, DrawerSlots } from './Drawer.types';
 export const renderDrawer_unstable = (state: DrawerState) => {
   const { slots, slotProps } = getSlots<DrawerSlots>(state);
 
-  // TODO Add additional slots in the appropriate place
+  if (state.type === 'temporary') {
+    return slots.dialog && slots.dialogSurface ? (
```

Comment:
is there a scenario where dialog or dialogSurface doesn't exist if type is temporary ? if that cannot happen with this type, we should probably throw an exception. wdyt?

Reply:
There is no scenario, but the Slots are optional. That's why the checks are needed here.

Reply:
gotcha, hmm.

so question from another angle. is there any scenario when user can/should override custom dialog/dialogSurface slots. if there is none - why do we need to provide this kind of "private api" ?